src/Resolver/Factory.php

```diff
@@ -62,4 +74,27 @@ protected function addPortToServerIfMissing($nameserver)
 
         return $nameserver;
     }
+
+    protected function determineSystemDefaultServer()
+    {
+        $nameservers = array();
+
+        if (stripos(PHP_OS, 'win') === false) {
+            $lines = file(self::UNIX_RESOLVE_CONF);
+            foreach ($lines as $line) {
+                $nameserverPosition = stripos($line, 'nameserver');
+                if ($nameserverPosition !== false) {
+                    $nameserverLine = trim(substr($line, $nameserverPosition + 11));
+                    list ($nameservers[]) = explode(' ', $nameserverLine);
+                }
+            }
+        }
+
+        if (count($nameservers) > 0) {
+            shuffle($nameservers);
+            return array_pop($nameservers);
+        }
+
+        throw new \Exception('Nameserver configuration missing');
```

Comment:
`man resolv.conf` says it should default to the local nameserver if none are present, but Amp doesn't do this either, at least currently.